### DISCUSSION TITLE
How to handle keydown events on canvas (when node not selected)?

### DISCUSSION THREAD

**Discussion by jeffgord** [xyflow/xyflow#4692] [Q&A] [answered]
Hi!

I'm looking for a way to handle keydown events on the flow canvas when there is no node selected. I have tried the `onKeyDown`, `onKeyDownCaptue`, and `OnKeyPress` props on the `<ReactFlow />` component, but these only work when the user has a node selected ([see this sandbox for a demo](https://codesandbox.io/p/sandbox/react-flow-canvas-onkeydown-ggy6p2?file=%2Fsrc%2FApp.tsx)). I have also tried wrapping `<ReactFlow />` in a `<div>` and using the `onKeyDown` prop of the div, but this does not work either.

I have seen solutions online where people attach a global event listener (with `window.addEventListener` or `document.addEventListener`), but this does not work for me as there are other elements on my page that I do not want the keydown to propagate to. Perhaps this involves some sort of conception of the whole flow element as being focusable, to distinguish keypresses on the canvas from keypresses on unrelated elements in the page.

Any ideas? Thanks so much for your help!

**Comment by bcakmakoglu** [comment, marked as answer]
Try setting `tabIndex={0}` on `<ReactFlow>` (or a wrapping `<div>`), that should allow you to listen to keyDown events.

[Here's an example.](https://codesandbox.io/p/sandbox/react-flow-canvas-onkeydown-forked-cm6mg6?file=%2Fsrc%2FApp.tsx%3A38%2C6)

**Reply by jeffgord** [reply]
Yup, this worked great - thanks for your help!
